File: src/foundation/java/io/karma/pda/foundation/component/Button.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.AbstractComponent;
import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 13/04/2024
 */
public class Button extends AbstractComponent {
    @Synchronize
    public final MutableState<String> text = MutableState.of("");

    public Button(final ComponentType<?> type, final UUID id) {
        super(type, id);
    }
}

File: src/composable/kotlin/io/karma/pda/composition/component/Button.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.Button
import io.karma.pda.foundation.component.DefaultComponents

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.Button(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable Button.() -> Unit = {}
) = component(
    type = DefaultComponents.BUTTON,
    layoutProps = layoutProps,
    localName = localName,
    props = props
)